### PR TITLE
Sockeye 2 training branch merge

### PR DESCRIPTION
**Pull request by mjdenkowski**
This PR merges in a cleaned up version of the sockeye_2_training branch.  Significant changes:

- Removed the `--weight-tying` arg and made `--weight-tying-type src_trg_softmax` the default.  This simplifies and clarifies our recommended setting.

- Retired BERTAdam.

- Added `none` as a value for `--learning-rate-scheduler` for completeness (train using a fixed learning rate).

- Minor Horovod version/settings update for faster training.

- Minor arg fixes: support different source/target values for transformer args, set AMP scale interval,  better parsing for optimizer params.

- Various minor edits for code and test cleanup.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test
- [x] Updated CHANGELOG.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/data_io.py [hunk 1]
```diff
@@ -1260,7 +1260,7 @@ def parallel_iterate(source_iterators: Sequence[Iterator[Optional[Any]]],
         "Different number of lines in source(s) and target iterables.")
 
 
-class FileListReader(Iterator):
+class FileListReader:
```
Comment (fhieber) [inline sockeye/data_io.py]: did you get typing errors or why did you remove this?

Comment (mjdenkowski) [reply sockeye/data_io.py]: Yes, newer versions of pylint were raising a warning:
```
E0239: Inheriting 'Iterator', which is not a class. (inherit-non-class)
```
